ph5/clients/ph5availability.py

```diff
@@ -384,27 +384,45 @@ def get_availability_extent(self, station='*', location='*',
                         if ret == -1:
                             continue
                         ph5_seed_station, ph5_loc, ph5_channel = ret
-
                         ph5_das = st['das/serial_number_s']
                         ph5_channum = st['channel_number_i']
-                        ph5_sample_rate = self.get_sample_rate(st)
-
-                        earliest, latest = self.ph5.get_extent(
-                            ph5_das, ph5_channum, ph5_sample_rate,
-                            starttime, endtime)
+                        ph5_start_epoch = st['deploy_time/epoch_l']
+                        ph5_stop_epoch = st['pickup_time/epoch_l']
+                        samplerate_return = None
+                        Das_t = self.ph5.query_das_t(
+                            ph5_das,
+                            chan=ph5_channum,
+                            start_epoch=ph5_start_epoch,
+                            stop_epoch=ph5_stop_epoch)
+                        for das in Das_t:
+                            ph5_sample_rate = das['sample_rate_i']
+                            earliest, latest = self.ph5.get_extent(
+                                ph5_das, ph5_channum, ph5_sample_rate,
+                                starttime, endtime)
+                            if das['sample_rate_i'] == st['sample_rate_i']:
+                                samplerate_return = das['sample_rate_i']
+                            if earliest is None or latest is None:
+                                continue
```

Comment:
This loop mess up sample rate at the result. It's ok if we don't include sample rate, but when we include sample rate, it gives the wrong result. Please look at station 0407 in array 2 and 3 and compare with das 5553 sample rate 100 and 200.

Reply:
The loop was slightly changed to make sure the correct sample rate is used when querying for cut times. 

``` 
                       for das in Das_t:
                            # Sample rates must first be compared than
                            if das['sample_rate_i'] == st['sample_rate_i']:
                                samplerate_return = das['sample_rate_i']
                                ph5_sample_rate = das['sample_rate_i']
                                times = self.ph5.get_availability(
                                        ph5_das, ph5_sample_rate, ph5_channum,
                                        starttime, endtime)
                            if times is None:
                                continue
```